consensus/propagation/have_wants.go

```diff
@@ -173,7 +171,12 @@ func (blockProp *Reactor) requestFromPeer(ps *PeerState) {
 						blockProp.Logger.Error("couldn't find proposal when filtering requests", "height", have.height, "round", have.round)
 						break
 					}
-					missingPartsCount = countRemainingParts(int(parts.Total()), len(parts.BitArray().GetTrueIndices()))
+				}
+
+				missingPartsCount := countRemainingParts(int(parts.Total()), len(parts.BitArray().GetTrueIndices()))
+				if missingPartsCount == 0 {
```

Comment:
this handles the case where the recovery parts were received between the time the haves were received and they were processed. So we no longer need to request any parts